### ISSUE TITLE
[C API] Scene Gets Cleared After Commit

### ISSUE THREAD

**Issue by anomiet** [ValveSoftware/steam-audio#251]
When loading a scene from disk with iplSceneLoad, and then calling iplSceneCommit on that scene anytime after it has loaded, the scene no longer sounds like the meshes exist anymore. It is as if the meshes are cleared from the scene. Running iplSceneLoad, then adding other meshes with iplInstancedMeshAdd or iplStaticMeshAdd, and finally calling iplSceneCommit, the geometry that was deserialized is gone, but the new geometry that was just committed is now there.

I was able to confirm this by resaving out the scene to objs after such calls to commit and seeing the geometry is gone. I can also hear that occlusion, pathing, and other such effects are no longer affected by the loaded geometry where they were before calls to commit.

**Comment by anomiet** [comment]
So I found a workaround by reading the source code for Unity integration and saw that the meshes are all combined into one IPLStaticMesh here:

https://github.com/ValveSoftware/steam-audio/blob/c6175a0b3c6c10d49bdd2887c76f7e17bb2bde14/unity/src/project/SteamAudioUnity/Assets/Plugins/SteamAudio/Scripts/Runtime/SteamAudioManager.cs#L1645

Doing this means I can skip the call to iplSceneLoad, and still easily serialize to one file for the whole scene while avoiding the issue I described in the original post.

From what I can tell, the Unity plugin never uses the iplSceneLoad functionality. There exists this constructor for Scene but I couldn't find it being invoked anywhere. I'm not going to close the issue cause it seems this is still not correct behavior for iplSceneLoad, but it can be avoided entirely. If it is intended, then docs should say so.
